src/database/schema.zmodel

```diff
@@ -797,3 +804,178 @@ model LastKeycloakUpdate {
+  availableTo DateTime? @map("available_to")
+
+
+  createdAt DateTime @default(now()) @map("created_at") @db.Timestamptz(6) @ignore()
+  updatedAt DateTime @default(now()) @map("updated_at") @db.Timestamptz(6) @updatedAt()
+  removedAt DateTime? @map("removed_at") @db.Timestamptz(6)
+
+  ticket Ticket?
+	consumable Consumable[]
+	questions ItemQuestion[]
+
+  // create: acces and mark self as author
+  @@allow("create", has(auth().policies, "webshop:create") && authorId == auth().memberId)
+  // update: author or manage access
+  @@allow("update", auth().memberId == authorId || has(auth().policies, "webshop:manage"))
+  @@allow("read", true)
+  // cannot read if removed, unless author or manager
+  @@deny("read", removedAt != null && !(auth().memberId == authorId || has(auth().policies, "webshop:manage")))
+  // only managers can delete (should very rarely be done)
+  @@allow("delete", has(auth().policies, "webshop:manage"))
+
+  @@map("shoppable")
+
+
+}
+
+model Ticket {
+  id String @id @db.Uuid
+  shoppable Shoppable @relation(fields: [id], references: [id], onDelete: Cascade, onUpdate: NoAction)
+
+  eventId String @db.Uuid
+  event Event @relation(fields: [eventId], references: [id], onDelete: NoAction, onUpdate: NoAction)
+
+
+  // same access as shoppable
+  @@allow("create", has(auth().policies, "webshop:create") && shoppable.authorId == auth().memberId)
+  @@allow("update", auth().memberId == shoppable.authorId || has(auth().policies, "webshop:manage"))
+  @@allow("read", true)
+  @@deny("read", shoppable.removedAt != null && !(auth().memberId == shoppable.authorId || has(auth().policies, "webshop:manage")))
+  @@allow("delete", has(auth().policies, "webshop:manage"))
+
+  @@map("ticket")
+}
+
+model ItemQuestion {
+  id String @id @default(dbgenerated("gen_random_uuid()")) @db.Uuid
+  shoppableId String @db.Uuid
+  shoppable Shoppable @relation(fields: [shoppableId], references: [id], onDelete: Cascade, onUpdate: NoAction)
+  title String
+  titleEn String?
+  description String
+  descriptionEn String?
+  type String @db.VarChar(255) // like "choice", "free-text" etc
+  forExternalsOnly Boolean @default(false) // if question should only be shown for people without an account
+	choices ItemQuestionAnswer[]
+	responses ItemQuestionResponse[]
+
+  // create: author of shoppable
+  @@allow("create", shoppable.authorId == auth().memberId)
+  @@allow("read", true)
+  // Unreadable if removed, unless author or manager
+  @@deny("read", shoppable.removedAt != null && !(auth().memberId == shoppable.authorId || has(auth().policies, "webshop:manage")))
+  // author (and managers) can update and delete questions
+  @@allow("update", auth().memberId == shoppable.authorId || has(auth().policies, "webshop:manage"))
+  @@allow("delete", auth().memberId == shoppable.authorId || has(auth().policies, "webshop:manage"))
+
+  @@map("item_question")
+}
+
+model ItemQuestionAnswer {
```

Comment:
I think we should choose a different name for this if it reflects the different options to choose between, I immediately got it mixed up with ItemQuestionResponse. `ItemQuestionVariety`or `ItemQuestionAlternative` is a lot better imo.

Reply:
Yeah I agree, naming is hard... What about `ItemQuestionOption`?

Reply:
That sounds good as well.